tests/test_bgp_neighbor_af.rb

```diff
@@ -230,9 +319,36 @@ def props_string(af, dbg)
   #   additional_paths_send
   #   soft_reconfiguration_in
   def test_tri_states
+    platform == :nexus ? tri_states_nexus : tri_states_ios_xr
+  end
+
+  def tri_states_ios_xr
     @@matrix.values.each do |af_args|
       af, dbg = clean_af(af_args)
+      %w(additional_paths_receive additional_paths_send).each do |k|
+        [:enable, :disable, :inherit, 'enable', 'disable', 'inherit'
+        ].each do |val|
+          assert_raises(UnsupportedError,
+                        "This feature #{k}, #{val} is not supported}") do
+            af.send("#{k}=", val)
+          end
+        end
+      end
+
+      %w(soft_reconfiguration_in).each do |k|
```

Comment:
it's the same apart from I raise an error and don't do an assert. I tried for a long time. I always ended up with some issue - as it's test code I decided to split it out.